iothub/service/src/Messaging/MessagingClient.cs

```diff
@@ -0,0 +1,379 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Transport types supported by MessagingClient.
+    /// </summary>
+    /// <remarks>
+    /// Amqp and Amqp over WebSocket only.
+    /// </remarks>
+    public enum TransportType
+    {
+        /// <summary>
+        /// Advanced Message Queuing Protocol transport.
+        /// </summary>
+        Amqp,
+
+        /// <summary>
+        /// Advanced Message Queuing Protocol transport over WebSocket only.
+        /// </summary>
+        Amqp_WebSocket_Only
+    }
+
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> for sending cloud to device and cloud to module messages.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d#message-feedback"/>.
+    public class MessagingClient : IDisposable
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly IotHubConnection _connection;
+        private readonly IotHubServiceClientOptions _clientOptions;
+        private readonly HttpClient _httpClient;
+        private readonly HttpRequestMessageFactory _httpRequestMessageFactory;
+        private readonly FaultTolerantAmqpObject<SendingAmqpLink> _faultTolerantSendingLink;
+        private readonly TimeSpan _openTimeout;
```

Comment:
We don't need default timeout values. If the user doesn't pass in a cancellation token, then they are fine with the method call potentially taking forever